### ISSUE TITLE
List all the tags with values

### ISSUE THREAD

**Issue by nkh** [oniony/TMSU#257]
I'm integrating TMSU in my file manager (maybe this time I'll tag for longer than 2 weeks before giving up)  and adding the possibility to select existing tags to apply on the selected files

tmsu tags doesn't report  "tag=value" but just "tag", this means that I have to first get all the tags, then get all the files that have the tag, then parse all the files tags till I get the one that has a value, if it has one, then extract the values. 

Can we, I please , get a tmsu tags --with-values, or something equivalent that wouldn't force one to go through hoops (that would be fine if it didn't take a noticeable time) or shall one hack the data out of the database?

Cheers

**Comment by oniony** [comment]
You can do `tmsu values TAG` to get the existing values for a tag. So you
could get all the tags with `tmsu tags` and look up the values for each
with `tmsu values TAG`. Not precisely what you want but not too bad.

**Comment by nkh** [comment]
certainly not too bad!

is there a way to get all the queries? I can "find" ~/mnt/queries but there may be a simpler way

**Comment by nkh** [comment]
and nice that one can give multiple tags to "tmsu values"!

**Comment by oniony** [comment]
Yeah, you should be able to do something like `tmsu tags | xargs tmsu
values`.

There's currently no way to query the queries from the CLI.

**Comment by oniony** [comment]
Yeah, can do, but it'll probably be in the Rust rewrite I'm planning and
I'm really busy with work at the moment so won't be for a while.

As an alternative you can query the database directly.

**Comment by nkh** [comment]
query the db directly is fine for me, could you please set me on the tracks?

**Comment by oniony** [comment]
The schema is detailed here:
https://github.com/oniony/TMSU/wiki/Database-Schema

You can install the Sqlite3 tooling and access the tmsu.db file directly
from the command-line or you can install any tool that has Sqlite3
connectivity. I'm quite fond of DBeaver for a GUI tool, for example.

From your app you would need a Sqlite3 provider.

Bear in mind the database schema could change from version to version. It's
very unlikely to considering I've not made any updates on a while. It'll
likely be different in the Rust rewrite.

**Comment by nkh** [comment]
Will the filesystem still be the same? in that case I may as well use it, it's just a few lines of code and speed is no problem for what I want to do. It's also fewer dependencies and I think it's nice to use core utilities.

**Comment by oniony** [comment]
I'm not sure yet.

**Comment by nkh** [comment]
OK, then, I'll get it out of the file system in the meantime, it's just  a few lines. Apropos file system, I had some difficulties getting to work as I expected, I got broken links and error messages about "Transport endpoint is not connected" and "deleted directories". I'll fill an issue when I'm sure I'm not the one doing something wrong.

I made a short video about the integration of tmsu in a file manager

https://drive.google.com/file/d/1BgTiiwqBTEy9jMGwYGPcMGe2jvoBodsN/view?usp=sharing
